app/components/SandboxSelector.tsx

```
@@ -69,26 +72,96 @@ export function SandboxSelector({
+      shortLabel: conn.osInfo?.hostname || conn.name,
+      icon: Laptop,
+    }),
+  );
   const options = [cloudOption, ...desktopOptions, ...remoteOptions];
 
+  // A connected Convex row can briefly exist before the command relay has
+  // subscribed. Confirm live Centrifugo presence before automatically choosing
+  // a remote runner over a reconnecting embedded bridge.
+  useEffect(() => {
+    if (!shouldVerifyRemotePresence) {
+      setOnlineRemoteConnectionIds(null);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/app/components/SandboxSelector.tsx:128:7
  126 |   useEffect(() => {
  127 |     if (!shouldVerifyRemotePresence) {
> 128 |       setOnlineRemoteConnectionIds(null);
      |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  129 |       return;
  130 |     }
  131 |

(react-hooks/set-state-in-effect)